cdap-master/pom.xml

```diff
@@ -157,24 +152,69 @@
+                  <goal>copy-resources</goal>
                 </goals>
-                <configuration>
-                  <outputDirectory>${stage.plugins.dir}</outputDirectory>
-                  <overWriteReleases>false</overWriteReleases>
-                  <overWriteSnapshots>false</overWriteSnapshots>
-                  <overWriteIfNewer>true</overWriteIfNewer>
-                  <includeArtifactIds>cdap-adapters</includeArtifactIds>
-                  <prependGroupId>true</prependGroupId>
-                  <silent>true</silent>
-                  <includeScope>compile</includeScope>
+                <configuration combine.self="override">
+                  <outputDirectory>${stage.opt.dir}/plugins</outputDirectory>
+                  <resources>
+                    <resource>
+                      <directory>${project.parent.basedir}/cdap-app-templates/cdap-etl/cdap-etl-batch/target</directory>
+                      <includes>
+                        <include>cdap-etl-batch-${project.version}.jar</include>
+                      </includes>
+                    </resource>
+                    <resource>
+                      <directory>
+                        ${project.parent.basedir}/cdap-app-templates/cdap-etl/cdap-etl-realtime/target
+                      </directory>
+                      <includes>
+                        <include>cdap-etl-realtime-${project.version}.jar</include>
+                      </includes>
+                    </resource>
+                  </resources>
+                </configuration>
+              </execution>
+              <execution>
+                <id>copy-realtime-plugin</id>
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>copy-resources</goal>
+                </goals>
+                <configuration combine.self="override">
+                  <outputDirectory>${stage.opt.dir}/plugins/${etl.realtime.plugin.name}</outputDirectory>
+                  <resources>
+                    <resource>
+                      <directory>${project.parent.basedir}/cdap-app-templates/cdap-etl/cdap-etl-lib/target</directory>
+                      <includes>
+                        <include>cdap-etl-lib-${project.version}.jar</include>
+                      </includes>
+                    </resource>
+                  </resources>
+                </configuration>
+              </execution>
+              <execution>
+                <id>copy-batch-plugin</id>
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>copy-resources</goal>
+                </goals>
+                <configuration combine.self="override">
+                  <outputDirectory>${stage.opt.dir}/plugins/${etl.batch.plugin.name}</outputDirectory>
+                  <resources>
+                    <resource>
+                      <directory>${project.parent.basedir}/cdap-app-templates/cdap-etl/cdap-etl-lib/target</directory>
```

Comment:
So etl lib JAR is copied to both realtime and batch? ok for now, do we need to split etl-lib into batch-lib, realtime-lib, transforms-lib?